src/main/java/jp/ac/keio/bio/fun/xitosbml/CuiMainSpatial.java

```diff
@@ -0,0 +1,167 @@
+
+	/**
+	 * The SpatialImage, which is a class for handling spatial image in XitoSBML.
+	 */
+	protected SpatialImage spImg;
+
+	/** TrialForImage **/
+	protected TrialForImg trial;
+
+	protected void cui(ImagePlus imager) {
+		hashDomainTypes = new HashMap<String, Integer>();
+		hashSampledValue = new HashMap<String, Integer>();
+		// imager.show();
+		trial = new TrialForImg(hashDomainTypes, hashSampledValue, imager);
+		// HashMap<String, ImagePlus> hashDomFile = trial.getDomFile();
+		// System.out.println(hashDomFile.values());
+	}
+
+	/**
+	 * Following process is performed to an image:
+	 * <ol>
+	 * <li>Interpolate an image if it is a Z-stack image (3D image) and the voxel
+	 * size of each axis (x, y and z) is not equal</li>
+	 * <li>Fill holes (blank pixels) in the image by morphology operation if
+	 * exists</li>
+	 * <li>Add a membrane between two different domains if exists</li>
+	 * </ol>
+	 *
+	 * The converted image will be generated as
+	 * {@link jp.ac.keio.bio.fun.xitosbml.image.SpatialImage}, which is a base class
+	 * for representing spatial image in XitoSBML.
+	 */
+
+	protected void computeImgTrial() {
+		Interpolator interpolator = new Interpolator();
+		HashMap<String, ImagePlus> hashDomFile = trial.getDomFile();
+		interpolator.interpolate(hashDomFile);
+		Filler fill = new Filler();
+
+		for (Entry<String, ImagePlus> e : hashDomFile.entrySet())
+			hashDomFile.put(e.getKey(), fill.fill(e.getValue()));
+
+		CreateImage creIm = new CreateImage(trial.getDomFile(), hashSampledValue);
+		spImg = new SpatialImage(hashSampledValue, hashDomainTypes, creIm.getCompoImg());
+		ImagePlus img = fill.fill(spImg);
+		spImg.setImage(img);
+		ImageBorder imgBorder = new ImageBorder(spImg);
+		spImg.updateImage(imgBorder.getStackImage());
+
+		new ImageEdit(spImg);
+
+	}
+
+	/**
+	 * Visualize the spatial model with ImageJ 3D Viewer.
+	 *
+	 * @param spImg the SpatialImage, which is a class for handling spatial image in
+	 *              XitoSBML.
+	 */
+	protected void visualize(SpatialImage spImg) {
+		viewer = new Viewer();
+		viewer.view(spImg);
+	}
+
+	/**
+	 * Create a tabbed table which allows users to add the SBases to the model
+	 * through GUI.
+	 */
+	protected void addSBases() {
+		ListOf<Parameter> lop = model.getListOfParameters();
+		ListOf<Species> los = model.getListOfSpecies();
+		TabTables tt = new TabTables(model);
+
+		while (tt.isRunning()) {
+			synchronized (lop) {
+				synchronized (los) {
+
+				}
+			}
+		}
+	}
+
+	/**
+	 * Show inclusion relationship of domains as a graph.
+	 */
+	protected void showDomainStructure() {
+		spatialplugin = (SpatialModelPlugin) model.getPlugin("spatial");
+		Geometry g = spatialplugin.getGeometry();
+		new DomainStruct().show(g);
+	}
+
+	/**
+	 * Visualize SpatialImage.
+	 *
+	 * @param spImg the SpatialImage, which is a class for handling spatial image in
+	 *              XitoSBML.
+	 */
+	protected void showStep(SpatialImage spImg) {
+		visualize(spImg);
+	}
```

Comment:
Do we need this part? This is relative to the GUI part of the code

Reply:
Will remove this.

Reply:
I am keeping showDomainStructure() and print() for verifying how the code works at this moment. Will remove these later on.

Reply:
Other functions pertaining to GUI were removed.

Reply:
Refers to https://github.com/Medha-B/XitoSBML/issues/4.